packages/frontend/src/pages/TokenSwap/utils/hooks/useCalculatedValues.js

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+
+import { decreaseByPercent, getPercentFrom } from '../../../../utils/amounts';
+
+export default function useCalculatedValues({
```

Comment:
I guess all of this could me moved to a sync function "getCalculatedValues" ?
I don't feel like we really need a custom hook for this.
Performance is not critical here, as no loops are involved. Just few basic arithmetic operations.